mllib/src/main/scala/org/apache/spark/ml/util/SchemaUtils.scala

```diff
@@ -213,9 +215,13 @@ private[spark] object SchemaUtils {
    */
   def getSchemaField(schema: StructType, colName: String): StructField = {
     val colSplits = AttributeNameParser.parseAttributeName(colName)
-    var field = schema(colSplits(0))
-    for (colSplit <- colSplits.slice(1, colSplits.length)) {
-      field = field.dataType.asInstanceOf[StructType](colSplit)
+    val sqlConf = SparkSession.getActiveSession.get.sessionState.conf
```

Comment:
We can simply do `SQLConf.get`, it also gets the conf object from the current active session.